src/embed_tests/TestInterrupt.cs

```diff
@@ -0,0 +1,83 @@
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+using NUnit.Framework;
+
+using Python.Runtime;
+
+namespace Python.EmbeddingTest
+{
+    public class TestInterrupt
+    {
+        private IntPtr _threadState;
+
+        [DllImport("Kernel32", EntryPoint = "GetCurrentThreadId", ExactSpelling = true)]
+        private static extern uint GetCurrentThreadId();
+
+        [DllImport("libc", EntryPoint = "pthread_self")]
+        private static extern IntPtr pthread_selfLinux();
+
+        [DllImport("pthread", EntryPoint = "pthread_self", CallingConvention = CallingConvention.Cdecl)]
+        private static extern ulong pthread_selfOSX();
+
```

Comment:
Python 3.8 has `threading.get_native_id()` for this, which we should probably expose in the library with a fallback to these.

Reply:
I am not sure how to implement this. I have added some changes, what do you think?